> `kfactory` should be updated in sync with `gdsfactor`, as the latter has a dependency on the former with tight version requirements.

Done.
But most `kfactory` tests are failing with:
```
tests/conftest.py:42: in lazy_datadir
    assert kf.config.project_dir is not None
E   AssertionError: assert None is not None
E    +  where None = Settings(n_threads=64, logfilter=LogFilter(level=<LogLevel.INFO: 'INFO'>, regex=None), display_type='image', meta_form...debug_names=False, write_cell_properties=True, write_context_info=True, write_file_properties=True, show_function=None).project_dir
E    +    where Settings(n_threads=64, logfilter=LogFilter(level=<LogLevel.INFO: 'INFO'>, regex=None), display_type='image', meta_form...debug_names=False, write_cell_properties=True, write_context_info=True, write_file_properties=True, show_function=None) = kf.config
```

--- a/pkgs/development/python-modules/gdsfactory/default.nix
+++ b/pkgs/development/python-modules/gdsfactory/default.nix
@@ -46,14 +46,14 @@
 }:
 buildPythonPackage rec {
   pname = "gdsfactory";
-  version = "9.18.1";
+  version = "9.20.1";
   pyproject = true;
 
   src = fetchFromGitHub {
     owner = "gdsfactory";
     repo = "gdsfactory";
     tag = "v${version}";
-    hash = "sha256-PPps3BaQbU7PCq+tlvjlPOurgBYHx/eGDEmlUmaB+O4=";
+    hash = "sha256-TpMi0Rv6sQA8uAPGl6iR1qgTU7havBlWmdz98DROoSk=";
   };
 
   build-system = [ flit-core ];

--- a/pkgs/development/python-modules/kfactory/default.nix
+++ b/pkgs/development/python-modules/kfactory/default.nix
@@ -12,7 +12,6 @@
   cachetools,
   klayout,
   loguru,
-  numpy,
   pydantic-extra-types,
   pydantic-settings,
   pydantic,
@@ -21,26 +20,32 @@
   rectangle-packer,
   requests,
   ruamel-yaml-string,
-  ruamel-yaml,
   scipy,
   semver,
   toolz,
   typer,
 
   # tests
+  pytest-regressions,
   pytestCheckHook,
 }:
 
 buildPythonPackage rec {
   pname = "kfactory";
-  version = "1.14.4";
+  version = "2.0.0";
   pyproject = true;
 
   src = fetchFromGitHub {
     owner = "gdsfactory";
     repo = "kfactory";
     tag = "v${version}";
-    hash = "sha256-el3bGv57mAfxYG9tdLX5N6R76F+9GY9jdZaIUjMqcVU=";
+    # kfactory uses `.git` to infer the project directory.
+    # https://github.com/gdsfactory/kfactory/blob/v2.0.0/src/kfactory/conf.py#L318-L327
+    # Otherwise, tests fail with:
+    # assert kf.config.project_dir is not None
+    # E   AssertionError: assert None is not None
+    leaveDotGit = true;
+    hash = "sha256-eZRNUb2Qw2HcR2W1pf15ulEt7ZCJwi60SuGdte/cG8E=";
   };
 
   build-system = [
@@ -48,12 +53,14 @@ buildPythonPackage rec {
     setuptools-scm
   ];
 
+  pythonRelaxDeps = [
+    "pydantic"
+  ];
   dependencies = [
     aenum
     cachetools
     klayout
     loguru
-    numpy
     pydantic
     pydantic-extra-types
     pydantic-settings
@@ -61,7 +68,6 @@ buildPythonPackage rec {
     rapidfuzz
     rectangle-packer
     requests
-    ruamel-yaml
     ruamel-yaml-string
     scipy
     semver
@@ -71,13 +77,67 @@ buildPythonPackage rec {
 
   pythonImportsCheck = [ "kfactory" ];
 
-  nativeCheckInputs = [ pytestCheckHook ];
+  nativeCheckInputs = [
+    pytest-regressions
+    pytestCheckHook
+  ];
+
+  disabledTests = [
+    # AssertionError: Binary files ... and ... differ
+    "test_array"
+    "test_array_indexerror"
+    "test_autorename"
+    "test_cell_default_fallback"
+    "test_cell_in_threads"
+    "test_cell_yaml"
+    "test_circular_snapping"
+    "test_create"
+    "test_enclosure_name"
+    "test_euler_snapping"
+    "test_filter_layer_pt_reg"
+    "test_filter_regex"
+    "test_flatten"
+    "test_info"
+    "test_invalid_array"
+    "test_kcell_attributes"
+    "test_namecollision"
+    "test_nested_dic"
+    "test_nested_dict_list"
+    "test_netlist"
+    "test_netlist_equivalent"
+    "test_no_snap"
+    "test_overwrite"
+    "test_ports_cell"
+    "test_ports_in_cells"
+    "test_ports_instance"
+    "test_rename_clockwise"
+    "test_rename_clockwise_multi"
+    "test_schematic_anchor"
+    "test_schematic_create"
+    "test_schematic_create_cell"
+    "test_schematic_kcl_mix_netlist"
+    "test_schematic_mirror_connection"
+    "test_schematic_route"
+    "test_size_info"
+    "test_to_dtype"
+  ];
 
   disabledTestPaths = [
     # https://github.com/gdsfactory/kfactory/issues/511
     "tests/test_pdk.py"
     # NameError
     "tests/test_session.py"
+
+    # AssertionError: Binary files ... and ... differ
+    "tests/test_all_angle.py"
+    "tests/test_cells.py"
+    "tests/test_grid.py"
+    "tests/test_l2n.py"
+    "tests/test_packing.py"
+    "tests/test_pins.py"
+    "tests/test_rename.py"
+    "tests/test_routing.py"
+    "tests/test_spiral.py"
   ];
 
   meta = {

